src/toxcore/friend_connection/packet/share_relays.rs

```diff
@@ -0,0 +1,82 @@
+/*! ShareRalays struct
+*/
+
+use toxcore::binary_io::*;
+use toxcore::state_format::old::*;
+
+/** ShareRelays is a struct that holds 3 TCP relays as a PackedNode format.
+
+This packet is used to share relay info. between two friends.
+
+Serialized form:
+
+Length    | Content
+--------- | ------
+`1`       | `0x11`
+variable  | 3 PackedNodes
+
+*/
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct ShareRelays {
+    relays: Vec<TcpUdpPackedNode>,
+}
+
+impl FromBytes for ShareRelays {
+    named!(from_bytes<ShareRelays>, do_parse!(
+        tag!("\x11") >>
+        relays: many1!(TcpUdpPackedNode::from_bytes) >>
+        verify!(value!(relays.len()), |len| len == 3) >>
+        (ShareRelays {
+            relays,
+        })
+    ));
+}
+
+impl ToBytes for ShareRelays {
+    fn to_bytes<'a>(&self, buf: (&'a mut [u8], usize)) -> Result<(&'a mut [u8], usize), GenError> {
+        do_gen!(buf,
+            gen_be_u8!(0x11) >>
+            gen_many_ref!(&self.relays, |buf, relay| TcpUdpPackedNode::to_bytes(relay, buf))
```

Comment:
Maybe check relays count? Like it was done here: https://github.com/tox-rs/tox/blob/e9c8174dbc804e167810589ab33f584c3ff448c5/src/toxcore/dht/packet/nodes_response.rs#L140

Reply:
ok, got it.

Reply:
Size of serialized array has nothing to do with our representation of packet struct. We can still have fixed size array since we know the count. It's not always convenient due to limitations of such arrays but maybe here we can use them. Or if not - then just add check for the _count_ (not length).

Reply:
The array size should be fixed when compile time. Right?
I think that array is not dynamic memory method.
The size of this struct is variable when run-time.

Reply:
But you wrote that there are always 3 nodes in this struct: https://github.com/tox-rs/tox/pull/205/files#diff-b046e67b27fca0a355ab477d4a703491R16

So why fixed array with a size of 3 can't be used?